### ISSUE TITLE
New Aerogarden App -> new API?

### ISSUE THREAD

**Issue by doctorkb** [dalinicus/homeassistant-aerogarden#36]
I just received notice from Aerogarden today that they're launching a new app.  Some of the older gardens (pre-2019) may not be compatible, which suggests to me they may be also revising the API.

Figured I'd let you know in case you didn't also receive the note.

More info here: https://aerogarden.com/learning/introducing-the-aerogarden-app.html

**Comment by dalinicus** [comment]
Hey, thanks for the heads up.  I did see that.

We did just move over to the v4 api with [1.4.2](https://github.com/dalinicus/homeassistant-aerogarden/releases/tag/1.4.2) after the certificate expired on an older endpoint.  So I'm guessing we should be good?

Just to be sure, I'll hook up the new app to a http proxy later tonight or this week and see if anything has changed.

**Comment by doctorkb** [comment]
Maybe we're in the clear, but this was concerning:
![image](https://github.com/dalinicus/homeassistant-aerogarden/assets/1383663/ac6c0fa2-c968-463d-9a65-bff4ec90c6ca)

-- it seems that if your garden works with the old app, it won't work with the new, and vice versa.

**Comment by dalinicus** [comment]
I think what they actually mean by this is that the old app has been disabled and no longer works at all.  Downloading the old app in a fresh emulator yields this message instead of the normal login.

![image](https://github.com/dalinicus/homeassistant-aerogarden/assets/8620219/a505292e-154c-41a3-a6f0-c50a94fa6802)

**Comment by doctorkb** [comment]
That makes sense... I'm seeing that now, too.

Also, I'm on 1.4.2 in HACS, but it seems to have broken overnight.  Reloading the integration doesn't fix it -- but the data is very clearly incorrect:
![image](https://github.com/dalinicus/homeassistant-aerogarden/assets/1383663/6dc43b51-5bf2-4d48-acef-ae6ed8c7d8be)

**Comment by dalinicus** [comment]
Yeah, it looks like app4.aerogarden.com is now returning old data, possibly when they cut over to the new system.  There's a new endpoint at app5.aerogarden.com with a completely different api backend... but I can't seem to get a proxy in between the app and the API because the new app doesn't appear to use the system certs for validation.  So I have no idea what the new calls look like.

I might need some help with someone that knows a bit more about android/ios development that might be able to figure out what calls its making so we can replicate it.

**Comment by doctorkb** [comment]
Sadly, I'm wondering if this isn't another Chamberlain MyQ situation...

**Comment by dalinicus** [comment]
honestly, same :(

**Comment by doctorkb** [comment]
> Chamberlain garbage door is still disconnected from Google Home

Pardon me, sir, but have you heard of our lord and savior ratgdo? https://paulwieland.github.io/ratgdo/#order 🙂

**Comment by samuel** [comment]
The new app uses flutter. This post discusses someone getting around the certificate pinning on Android using Ghidra to patch the library: https://raphaeldenipotti.medium.com/bypassing-ssl-pinning-on-android-flutter-apps-with-ghidra-77b6e86b9476

I don't have time at the moment to try it, but throwing it out there in case  someone wants to explore this option.
